src/futils.c

```diff
@@ -21,32 +21,41 @@ int git_futils_mkpath2file(const char *file_path, const mode_t mode)
 		GIT_MKDIR_PATH | GIT_MKDIR_SKIP_LAST | GIT_MKDIR_VERIFY_DIR);
 }
 
+static void git_buf_put_rand(git_buf *buf, size_t chars)
```

Comment:
This probably belongs in `buffer.c`.

Reply:
I was hesitant to put it in buffer.c because the "randomness" is really bad, but there doesn't seem to be any existing abstraction for randomness. It's probably mostly fine in this case, because we're opening the file with the precise permissions we want, so the worst that happens is that if multiple processes try to create temp files, a few of them will have to retry a few times, but I didn't want to expose this function to users that would expect actual randomness.

Reply:
It’s still best to put it in with the rest of the buffer code from an organizational standpoint.  That’s all internal to the library and I don’t think anyone will go assuming that it’s crypto-worthy.

Having said that, we may want to introduce a random number generator for the library that produces better randomness but `rand` is so meh that it might be nice.

Reply:
It’s still best to put it in with the rest of the buffer code from an organizational standpoint.  That’s all internal to the library and I don’t think anyone will go assuming that it’s crypto-worthy.

Having said that, we may want to introduce a random number generator for the library that produces better randomness but `rand` is so meh that it might be nice.

Reply:
Done.